packages/cozy-viewer/src/helpers.js

```
@@ -84,6 +84,22 @@ export const roughTokensEstimation = text => {
+ * Sanitize text, typically from after file text extraction
+ * @param {string} text - The text to sanitize
+ * @returns {string} The sanitized text
+ */
+export const sanitizeText = text => {
+  return (
+    text
+      .replace(/\r\n?/g, '\n') // Normalize line endings
+      .replace(/[^\S\r\n]{4,}/g, ' ') // Compress spaces
+      .replace(/\n{2,}/g, '\n') // Remove empty lines
+      // eslint-disable-next-line no-control-regex
+      .replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, '') // Remove control chars
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)